packages/api-client/src/api-client.ts

```diff
@@ -239,10 +241,10 @@ export class GFWAPI {
         const data = await fetch(fetchUrl, {
           method,
           signal,
-          ...(body && { body: JSON.stringify(body) }),
+          ...(requestType === 'json' ? body && { body: JSON.stringify(body) } : body && { body }),
           headers: {
             ...headers,
-            ...(responseType === 'json' && { 'Content-Type': 'application/json' }),
+            ...(requestType === 'json' && { 'Content-Type': 'application/json' }),
```

Comment:
So the `Content-Type` is now dependent of the `requestType` only, is that correct? By looking at the function signature it can receive both `requestType` and `responseType` and they could not match. What would be the expected header to send in the request when `responseType = 'json'` and `requestType = 'formData'`? Is this a valid scenario?

Reply:
@rdgfuentes the response type is for this:
.then((res) => {
            switch (responseType) {
              case 'default':
                return res
              case 'json':
                return parseJSON(res)
              case 'blob':
                return res.blob()
              case 'text':
                return res.text()
              case 'arrayBuffer':
                return res.arrayBuffer()
              case 'vessel': {
                try {
                  return import('@globalfishingwatch/pbf/decoders/vessels.js').then(